### PR TITLE
V-USB Interface reorder

### PR DESCRIPTION
**Pull request by yiancar**
This might seem irrelevant now as we only got 2 endpoints, but with a 3rd endpoint (https://github.com/qmk/qmk_firmware/pull/9020) it is needed.

I have PRed in master as this is NOT breaking change.

## Description

- Reordered usb_interface enumeration.
- VIA expects RAW_HID endpoint to be at interface 1.
- This does not break anything with only 2 endpoints. Keyboard is always at interface 0 and the extra feature is at interface 1.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by yiancar** [comment]
I think the technical change your proposing is more generalized than V-USB.
Chibios and LUFA use: https://github.com/qmk/qmk_firmware/blob/805f5cb72bb04c10327b40ed0c41e3848ad9ca75/tmk_core/protocol/usb_descriptor.h#L131 which similarly is executed in the same way as the change proposed here.

TL;DR: this PR aligns the interface behavior as the other 2 USB "libraries".

File: tmk_core/protocol/vusb/vusb.c [hunk 1]
```diff
@@ -43,12 +43,12 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 enum usb_interfaces {
     KEYBOARD_INTERFACE = NEXT_INTERFACE,
-#if (defined(MOUSE_ENABLE) || defined(EXTRAKEY_ENABLE))
-    MOUSE_EXTRA_INTERFACE = NEXT_INTERFACE,
-#endif
 #ifdef RAW_ENABLE
```
Comment (zvecr) [inline tmk_core/protocol/vusb/vusb.c:46]:
```suggestion
// It is important that the Raw HID interface is at a constant
// interface number, to support Linux/OSX platforms and chrome.hid
// If Raw HID is enabled, let it be always 1.
#ifdef RAW_ENABLE
```

Comment (yiancar) [reply tmk_core/protocol/vusb/vusb.c:46]: It has to do with limitations of hidapi. On some platforms you cannot address an endpoint without its interface.

**Comment by yiancar** [comment]
Done

**Comment by yiancar** [comment]
It did before anyway sadly
